configs/components/pl-ruby-patch.rb

```diff
@@ -61,9 +61,16 @@
       sed_command = %(s|Gem.ruby.shellsplit|& << '-r/opt/puppetlabs/puppet/share/doc/rbconfig-#{settings[:ruby_version]}-orig.rb'|)
     end
 
+    # rubygems switched which file has the command we need to patch starting with ruby 3.2.2
+    if Gem::Version.new(settings[:ruby_version]) >= Gem::Version.new('3.2.2')
```

Comment:
I'm not sure of a better way to check for semantic versioning, but this is what a Google search suggested and it works.

Reply:
:+1: we use that in puppet too